Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonAnimationManager.cs

```diff
@@ -0,0 +1,149 @@
+/******************************************************************************
+// File Name:       HarpoonAnimationManager.cs
+// Author:          Andrew Stapay
+// Creation Date:   November 18, 2024
+//
+// Description:     Assists with the logic of the harpoon's animations.
+******************************************************************************/
+using System.Collections;
+using UnityEngine;
+
+public class HarpoonAnimationManager : MonoBehaviour
+{
+    private Animator _animator;
+
+    // The names of the parameters used in the Animator
+    private const string _FIRE_ANIM = "shoot";
+    private const string _FOCUS_START_ANIM = "holdFocus";
+    private const string _FOCUS_END_ANIM = "releaseFocus";
+    private const string _RELOAD_READY_ANIM = "reloadReady";
+    private const string _AMMO_EMPTY_ANIM = "ammoEmpty";
+    private const string _NOT_WALL_ANIM = "notAtWall";
+    private const string _AT_WALL_ANIM = "atWall";
+
+    //Used to check to see if we are near a wall
+    private float _WALL_CHECK_DIST = 1;
+
+    /// <summary>
+    /// Called when the game starts
+    /// Used to get the animator, subscribe to events, and start the coroutine for checking for walls.
+    /// </summary>
+    private void Awake()
+    {
+        _animator = GetComponent<Animator>();
+        SubscribeToEvents();
+        StartCoroutine(CheckForWall());
```

Comment:
Personally I like to keep awake/start clear of anything other than function calls. This is not a required change